common/src/main/kotlin/com/hartwig/actin/molecular/evidence/actionability/IndirectEvidenceMatcher.kt

```diff
@@ -0,0 +1,107 @@
+    }
+
+    private fun findIndirectEvidence(gene: String, proteinEffect: ProteinEffect): Set<EfficacyEvidence> {
+        return treatmentByGeneEffectForHotspots[TreatmentKey(gene, proteinEffect)] ?: emptySet()
+    }
+
+    companion object {
+
+        val logger: Logger = LogManager.getLogger(IndirectEvidenceMatcher::class.java)
+
+        fun create(serveRecord: ServeRecord): IndirectEvidenceMatcher {
+            val knownHotspotsByPosition = mapKnownHotspotsByPosition(serveRecord)
+
+            val treatmentByGeneEffect = serveRecord.evidences()
+                .asSequence()
+                .filter { it.source() == Knowledgebase.CKB }
+                .filterNot { isCombinedProfile(it.molecularCriterium()) }
+                .filter { it.molecularCriterium().hotspots().any() }
+                .mapNotNull { evidence -> matchableEvidence(evidence, knownHotspotsByPosition) }
+                .groupBy({ it.first }, { it.second })
+                .mapValues { (_, evidences) -> evidences.toCollection(HashSet()) }
+
+            return IndirectEvidenceMatcher(treatmentByGeneEffect)
+        }
+
+        fun empty(): IndirectEvidenceMatcher = IndirectEvidenceMatcher(emptyMap())
+
+        private fun mapKnownHotspotsByPosition(serveRecord: ServeRecord): Map<HotspotKey, KnownHotspot> {
+            val groupedByPosition = serveRecord.knownEvents().hotspots()
+                .asSequence()
+                .filter { it.sources().contains(Knowledgebase.CKB) }
+                .filterNot { it.associatedWithDrugResistance() == true }
+                .groupBy { it.toKey() }
+
+            groupedByPosition
+                .filter { (_, hotspots) -> hotspots.size > 1 }
+                .forEach { (key, hotspots) ->
+                    hotspots.drop(1).forEach { duplicate ->
+                        logger.warn("KnownHotspot with key $key already exists in map; skipping duplicate for known hotspot $duplicate.")
+                    }
+                }
+
+            return groupedByPosition.mapValues { (_, hotspots) -> hotspots.first() }
+        }
+
+        private fun matchableEvidence(
+            evidence: EfficacyEvidence,
+            knownHotspotsByVariant: Map<HotspotKey, KnownHotspot>
+        ): Pair<TreatmentKey, EfficacyEvidence>? {
+            return evidence.molecularCriterium().hotspots().firstOrNull()
+                ?.takeIf { ApplicabilityFiltering.isApplicable(it) }
+                ?.variants()?.firstOrNull()
+                ?.let { variant -> knownHotspotsByVariant[variant.toKey()] }
+                ?.takeIf { hotspot -> hotspot.associatedWithDrugResistance() != true }
+                ?.takeIf { GenericInhibitorFiltering.isGenericInhibitor(evidence.treatment()) }
+                ?.let { knownHotspot ->
+                    val proteinEffect = GeneAlterationFactory.convertProteinEffect(knownHotspot.proteinEffect())
+                    TreatmentKey(knownHotspot.gene(), proteinEffect) to evidence
+                }
```

Comment:
We should only match on `GAIN_OF_FUNCTION` (can include "predicted") and `LOSS_OF_FUNCTION` (can include "predicted"). Maybe that filtering could be added here?